onadata/apps/viewer/models/data_dictionary.py

```diff
@@ -78,9 +78,17 @@ def set_uuid_in_xml(self, file_name=None, id_string=None):
         doc = clean_and_parse_xml(self.xml)
         model_nodes = doc.getElementsByTagName("model")
         if len(model_nodes) != 1:
-            raise Exception("xml contains multiple model nodes")
-
-        model_node = model_nodes[0]
+            node_counts = 0
+            for node in model_nodes:
+                if hasattr(node, 'firstChild') and hasattr(node.firstChild, 'nodeName'):
+                    if node.firstChild.nodeName == 'instance':
+                        if node_counts != 0:
+                            raise Exception("xml contains multiple model nodes")
+                        else:
+                            model_node = node
+                            node_counts += 1
+        else:
```

Comment:
As we discussed, this code works to match @jnm's [first comment](https://github.com/kobotoolbox/kobocat/issues/486#issuecomment-832266833)

> I'm guessing a <model> that's a child (or further descendant) of <instance> is just fine:
https://getodk.github.io/xforms-spec/#structure.

But as John said right after, it's better to search for immediate children of `<h:head>` 

> look for <model> elements only that are the immediate children of <h:head>